baize/responses.py

```diff
@@ -178,16 +178,19 @@ def parse_range(
             raise MalformedRangeHeader("Only support bytes range")
 
         ranges = [
-            (int(_[0]), int(_[1]) + 1 if _[1] else max_size)
-            for _ in re.findall(r"(\d+)-(\d*)", ranges_str)
+            (
+                int(_[0]) if _[0] else max_size - int(_[1]),
+                int(_[1]) + 1 if _[0] and _[1] and int(_[1]) < max_size else max_size,
```

Comment:
Parsing `-` will got an error……We should fix the regex or judgement.

Reply:
Good catch, I hadn't considered that case. I believe the correct response would be to return a `MalformedRangeHeader`. I will address this.

Reply:
Looking at the existing logic and reviewing the RFC, I think the correct answer may be to ignore it. It already ignores any section of the header that doesn't match the regex (e.g. `0-10,hello world` is valid for 1-10).

However, I think that a case where parse_range would return an empty range should result in a `MalformedRangeHeader`.